swap/simulations_test.go

```diff
@@ -0,0 +1,695 @@
+	}
+	if cheque2, err = creditorSvc.swap.loadLastReceivedCheque(debitor); err != nil {
+		t.Fatalf("expected cheques with debitor, but none found")
+	}
+
+	// both cheques (at issuer and beneficiary) should have same cumulative value
+	if cheque1.CumulativePayout != cheque2.CumulativePayout {
+		t.Fatalf("Expected symmetric cheques payout, but they are not: %d vs %d", cheque1.CumulativePayout, cheque2.CumulativePayout)
+	}
+
+	// check also the actual expected amount
+	expectedPayout := uint64(maxCheques * (DefaultPaymentThreshold + 1))
+
+	if cheque2.CumulativePayout != expectedPayout {
+		t.Fatalf("Expected %d in cumulative payout, got %d", expectedPayout, cheque1.CumulativePayout)
+	}
+
+	log.Info("Simulation ended")
+}
+
+// TestBasicSwapSimulation starts 16 nodes, then in a simple round robin fashion sends messages to each other.
+// Then checks that accounting is ok. It checks the actual amount of balances without any cheques sent,
+// in order to verify that the most basic accounting works.
+func TestBasicSwapSimulation(t *testing.T) {
+	nodeCount := 16
+	// create the shared backend and params
+	params, err := newSharedBackendSwaps(nodeCount)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// cleanup backend
+	defer params.backend.Close()
+
+	// initialize the simulation
+	sim := simulation.NewBzzInProc(newSimServiceMap(params), false)
+	defer sim.Close()
+
+	log.Info("Initializing")
+
+	ctx, cancelSimRun := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancelSimRun()
+
+	_, err = sim.AddNodesAndConnectFull(nodeCount)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	log.Debug("Wait for all connections to be established")
+
+	simulations.VerifyFull(t, sim.Net, sim.NodeIDs())
+
+	log.Info("starting simulation...")
+
+	// we don't want any cheques to be issued for this test, we only want to test accounting across nodes
+	// for this we define a "global" maximum amount of messages to be sent;
+	// this formula should ensure that we trigger enough messages but not enough to trigger cheques
+	maxMsgs := (DefaultPaymentThreshold / params.maxMsgPrice) * (nodeCount - 1)
+
+	// need some synchronization to make sure we wait enough before checking all balances:
+	// all messages should have been received, otherwise there may be some imbalances!
+	allMessagesArrived := make(chan struct{})
+
+	metricsReg := metrics.AccountingRegistry
+	cter := metricsReg.Get("account.msg.credit")
+	counter := cter.(metrics.Counter)
+	counter.Clear()
+
+	go func() {
+		for {
+			maxMsgsInt64 := int64(maxMsgs)
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+			// all messages have been received
+			if counter.Count() == maxMsgsInt64 {
+				close(allMessagesArrived)
+				return
+			}
+			time.Sleep(10 * time.Millisecond)
+		}
+	}()
+
+	result := sim.Run(ctx, func(ctx context.Context, sim *simulation.Simulation) (err error) {
+		log.Info("simulation running")
+
+		ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
```

Comment:
There already is a context passed in this scope created on line 467. This can be removed.